src/datasource.js

```diff
@@ -59,6 +59,19 @@ export class GenericDatasource {
     return this.performTimeseriesQuery(options);
   }
 
+  annotationQuery(options) {
+    const query = this.createLogsQueryForAnnotation(options);
+    return this.backendSrv.datasourceRequest(query)
+      .then( (response) => {
+        const data = response.data;
+        const timeField = options.annotation.timeField || "timestamp"
+        const timeEndField = options.annotation.timeEndField || null
+        const textField = options.annotation.textField || "message"
+        return GenericDatasource.transformAnnotationResults(data.matches, timeField, timeEndField, textField);
```

Comment:
I hadn't thought of this case, but having just tried it it throws up an error in the Grafana UI with the proper error message received from Scalyr. I believe the proxy layer Grafana has for interacting with datasources has error handling like this. Honestly this seems like the optimal thing to do, so I think we should leave it as is, unless you think otherwise?

Reply:
Just to add this here for reference / context, since I was also curious - https://grafana.com/docs/grafana/latest/packages_api/runtime/backendsrv/

> By default Grafana will display an error message alert if the remote call fails. If you want to prevent this from happending you need to catch the error thrown by the BackendSrv and set the showErrorAlert = true on the request options object.

So yeah, it looks like that as long as API correctly returns non-200 status code we should be fine (as long as we want to directly show the error returned by the API)

Reply:
I did verify the Scalyr api return non-200.